Spigot-Server-Patches/0299-AnvilDamageEvent.patch

```diff
@@ -0,0 +1,30 @@
+From 1157c45da93ef634465a4f8b5c0a3e651392a79a Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Fri, 20 Jul 2018 23:37:03 -0500
+Subject: [PATCH] AnvilDamageEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/ContainerAnvil.java b/src/main/java/net/minecraft/server/ContainerAnvil.java
+index c252ff6d5..8fc4d9249 100644
+--- a/src/main/java/net/minecraft/server/ContainerAnvil.java
++++ b/src/main/java/net/minecraft/server/ContainerAnvil.java
+@@ -73,6 +73,16 @@ public class ContainerAnvil extends Container {
+                 if (!world.isClientSide) {
+                     if (!entityhuman.abilities.canInstantlyBuild && iblockdata.a(TagsBlock.x) && entityhuman.getRandom().nextFloat() < 0.12F) {
+                         IBlockData iblockdata1 = BlockAnvil.a_(iblockdata);
++                        // Paper start
++                        com.destroystokyo.paper.event.block.AnvilDamagedEvent event = new com.destroystokyo.paper.event.block.AnvilDamagedEvent(getBukkitView(), org.bukkit.craftbukkit.block.data.CraftBlockData.fromData(iblockdata1));
++                        if (!event.callEvent()) {
++                            return itemstack;
++                        } else if (event.getDamageState() == com.destroystokyo.paper.event.block.AnvilDamagedEvent.DamageState.BROKEN) {
++                            iblockdata1 = null;
++                        } else {
++                            iblockdata1 = ((org.bukkit.craftbukkit.block.data.CraftBlockData) event.getDamageState().getMaterial().createBlockData()).getState();
```

Comment:
This is going to cause it to lose its facing data. We need to carry over previous facing data, see BlockAnvil.a_()

Reply:
Fixed.